tests/MonitoringFunctions.Test/TestHelperMethods.cs

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using MonitoringFunctions.Providers;
+using System.Threading.Tasks;
+
+namespace MonitoringFunctions.Test
+{
+    [TestClass]
+    public class TestHelperMethods
+    {
+        [TestMethod]
+        public async Task TestCheckUrlAccess()
+        {
+            using IDataService dataService = new DummyDataService();
+            await HelperMethods.CheckAndReportUrlAccess(NullLogger.Instance, "test_run", "https://www.microsoft.com", dataService);
+        }
+
+        [DataRow("definitely not a url")]
+        [DataRow("https://www.somewebsitethatdontexist1239420312392304.com")]
```

Comment:
This URL may fail this test one time (highly unlikely, but still). Maybe use instead some reserved URL (like 0.com)

Reply:
Thanks, I didn't know about 0.com. Is it a reserved URL or a an invalid one? Can domain names begin with a number?

Reply:
The specification doesn't says anything about that (https://tools.ietf.org/html/rfc3986#section-3.2.2). When I looked up the domain in the [whois tool](https://lookup.icann.org/lookup) it said that it is registered under IANA.

Reply:
Awesome!